tests/test_pulp_clear_repos.py

```diff
@@ -0,0 +1,872 @@
+        self.detail_client_mock(PDCClientClassMock, PulpAdminConfigClassMock, expected, self.expected_query, expected_query_add, nameFunction, result, commit)
+
+    @patch('releng_sop.pulp_clear_repos.PDCClient', autospec=True)
+    @patch('releng_sop.pulp_clear_repos.PulpAdminConfig', autospec=True)
+    def test_details_arch_variant_with_commit_more_repo(self, PulpAdminConfigClassMock, PDCClientClassMock):
+        """Get details when commiting."""
+        self.data['repo_family'] = 'htb'
+        self.data['arch'] = ['x86_64']
+        self.data['variant_uid'] = ['Server']
+
+        result = [
+            {
+                'name': 'rhel-7-workstation-htb-rpms\n     rhel-7-server-htb-source-rpms',
+            }
+        ]
+
+        expected = self.details_base + self.details_good_repo + self.details_variant_arch + self.details_with_more_repo
+        expected_query_add = {
+            'arch': self.data['arch'],
+            'variant_uid': self.data['variant_uid'],
+        }
+
+        commit = True
+        nameFunction = TestPulpClearRepos.test_details_arch_variant_with_commit_more_repo
+        self.detail_client_mock(PDCClientClassMock, PulpAdminConfigClassMock, expected, self.expected_query, expected_query_add, nameFunction, result, commit)
+
+    @patch('releng_sop.pulp_clear_repos.PDCClient', autospec=True)
+    @patch('releng_sop.pulp_clear_repos.PulpAdminConfig', autospec=True)
+    def test_details_arch_variant_with_commit_no_repo(self, PulpAdminConfigClassMock, PDCClientClassMock):
+        """Get details when commiting."""
+        self.data['repo_family'] = 'ht'
+        self.data['arch'] = ['x86_64']
+        self.data['variant_uid'] = ['Server']
+
+        result = [
+            {
+                'name': 'No repos found.',
+            }
+        ]
+
+        expected = self.details_base + self.details_bad_repo + self.details_variant_arch + self.details_no_repo
+        expected_query_add = {
+            'arch': self.data['arch'],
+            'variant_uid': self.data['variant_uid'],
+        }
+
+        commit = True
+        nameFunction = TestPulpClearRepos.test_details_arch_variant_with_commit_no_repo
+        self.detail_client_mock(PDCClientClassMock, PulpAdminConfigClassMock, expected, self.expected_query_no_repo, expected_query_add, nameFunction, result, commit)
+
+    def share_get_cmd(self, PulpAdminConfigClassMock, expected, commit, nameFunction):
+        """Check the expected and actual."""
+        clear = PulpClearRepos(self.env, self.release, self.data['repo_family'], self.data['variant_uid'], self.data['arch'])
+        clear.repos = self.repos
+        pulpAdminConfig = PulpAdminConfigClassMock.return_value
+        pulpAdminConfig.name = 'pulp-test'
+        pulpAdminConfig.config_path = 'some_path.json'
+
+        client = pulpAdminConfig.__getitem__.return_value
+        client.__getitem__.return_value = 'admin'
+        if commit:
```

Comment:
I am not sure if this `if... else...` is needed, `assertEqual` should be able to compare list of list of string. The idea is that if possible the output to be checked by the test should not be altered by the test, rather should be used as it is.